src/php/types/object.rs

```diff
@@ -352,31 +417,43 @@ impl<'a, T: RegisteredClass> FromZval<'a> for &'a T {
     const TYPE: DataType = DataType::Object(Some(T::CLASS_NAME));
 
     fn from_zval(zval: &'a Zval) -> Option<Self> {
-        let cobj = ZendClassObject::<T>::from_zend_obj_ptr(zval.object()?)?;
+        let cobj = ZendClassObject::<T>::from_zend_obj(zval.object()?)?;
 
-        Some(unsafe { &*cobj.obj.as_mut_ptr() })
+        Some(unsafe { &*cobj.obj.as_ptr() })
     }
 }
 
-impl<'a, T: RegisteredClass> FromZval<'a> for &'a mut T {
-    const TYPE: DataType = DataType::Object(Some(T::CLASS_NAME));
+// TODO(david): mutable FromZval?
+// impl<'a, T: RegisteredClass> FromZval<'a> for &'a mut T {
```

Comment:
Because `Zval::object()` was changed to return an immutable reference, `FromZval` can no longer be implemented for `&mut T`. I think we will need to add a new trait `FromZvalMut`, and then any type which implements `FromZval` can implement `FromZvalMut`.

Reply:
Perhaps this should be done in a later PR as it will require keeping a mutable `Zval` inside `Arg`